staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1113,7 +1127,19 @@ func (e *Store) Delete(ctx context.Context, name string, deleteValidation rest.V
 	}
```

Comment:
if the unsafe deletion flow is mingled with the graceful deletion flow we have today, it might be harder to follow the code, i think having the logic for unsafe deletion co-located in a separate file makes it more easy to follow. 

A couple of points:
- If we want to know whether graceful Deletion has failed, the unsafe deleter needs to wrap the graceful deleter
- the way i am wiring the unsafe deleter is from the `DeleteResource` endpoint handler. The unsafe deleter is used if and only if the user has enabled the delete option.

Reply:
it's easier to wire the unsafe-deleter from inside if we have it as a separate unit, rather than making the changes inline
```
		if utilfeature.DefaultFeatureGate.Enabled(features.AllowUnsafeMalformedObjectDeletion) {
			if options != nil && ptr.Deref(options.IgnoreStoreReadErrorWithClusterBreakingPotential, false) {
				p, ok := r.(rest.CorruptObjectDeleterProvider)
				if !ok {
					// TODO: this is a developer error
					scope.err(fmt.Errorf("ignoreStoreReadErrorWithClusterBreakingPotential is not supported"), w, req)
					return
				}
				r = p.GetCorruptObjDeleter()
			}
		}
```